examples/cluster/plot_cluster_comparison.py

```diff
@@ -35,33 +34,46 @@
 from sklearn.preprocessing import StandardScaler
 from itertools import cycle, islice
 
-np.random.seed(0)
 
-# ============
-# Generate datasets. We choose the size big enough to see the scalability
-# of the algorithms, but not too big to avoid too long running times
-# ============
+# %%
+# Generate sample data
+# ---------------------------------
+
+# We choose the size big enough to see the scalability
+# of the algorithms, but not too big to avoid too long running times.
+# Sample data consists of
+# 1. noisy circles 2. noisy moons 3. blobs with varied variance
+# 4. anisotrophic distributions ('stripes') 5. blobs with smaller variance
+# 6. rectangular noise
+
 n_samples = 500
+# 1. make large circle containing smaller circle
 noisy_circles = datasets.make_circles(n_samples=n_samples, factor=0.5, noise=0.05)
+# 2. make two half circles
 noisy_moons = datasets.make_moons(n_samples=n_samples, noise=0.05)
-blobs = datasets.make_blobs(n_samples=n_samples, random_state=8)
-no_structure = np.random.rand(n_samples, 2), None
-
-# Anisotropicly distributed data
-random_state = 170
-X, y = datasets.make_blobs(n_samples=n_samples, random_state=random_state)
+# 3. blobs with varied variances
+rng = np.random.RandomState(0)
```

Comment:
Instead of passing the random number generator, I would prefer to pass an integer in each function with `random_state=0`.
It is not the best practice to pass the same random number generator so we can avoid to do that if we rework the example.